model/host/host.go

```diff
@@ -2663,6 +2663,22 @@ func (h *Host) MarkShouldExpire(expireOnValue string) error {
 	)
 }
 
+// UnsetHomeVolume disassociates a home volume from a (stopped) host.
+// This is for internal use, and should only be used on hosts that
+// will be terminated imminently; otherwise, the host will fail to boot.
+func (h *Host) UnsetHomeVolume() error {
+	err := UpdateOne(
+		bson.M{IdKey: h.Id},
+		bson.M{"$unset": bson.M{HomeVolumeIDKey: true}},
```

Comment:
Since the BSON tags for `HomeVolumeID` is not `omitempty`, the data model implies that the field should always be set, even if it's empty. Should this field instead be set to the empty string?

Reply:
Based on the unit test it seems like these have the same effect? But `$set` does seem more correct, updated!

Reply:
I think unset is a bit clearer personally, and is more correct for moving to an empty state; i think omitempty just causes it to default.

Reply:
(I see this was already merged so never mind 😅 but a future thought)

Reply:
I guess most of the difference here is how you query for a zero vlaue field later. Unset will delete the field from the doc (so you would query for it with `{$exists: false}`). Omitempty will unset if it's the Go zero value (so you would query for it with {"home_volume_id": ""}`. Since the field isn't `omitempty`, the Go driver will always set it to `""` if unspecified, unless you write a query that explicitly `$unset`s it. It seems easier to just have it be consistent with the data model and not have to query both for existence and for the empty string.